.github/workflows/tarball-check.yml

```diff
@@ -0,0 +1,52 @@
+name: Check Tarball
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK 11
+        uses: actions/setup-java@v4
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+          cache: 'sbt'
+      - name: Build tarball
```

Comment:
This is fairly detailed logic for tests and it should not written in Github action. One should be able to run these tests locally. So this entire test logic needs to be `tests/test-tarball-generation.sh/py"